src/public-api-types.tests.ts

```diff
@@ -0,0 +1,216 @@
+/**
+ * Type tests (run them with yarn test:types)
+ */
+import { folder } from './helpers'
+import { useControls } from './useControls'
+
+declare function expectType<T>(
+  v: T
+): void
+
+/**
+ * numbers
+ */
+;(() => {
+  expectType<{ a: number }>(useControls({ a: 10 }))
+  expectType<{ a: number }>(useControls({ a: { value: 10 } }))
+  expectType<{ a: number }>(useControls({ a: { value: 10, min: 0 } }))
+  expectType<{ a: number }>(useControls({ a: { value: 10, min: 0, max: 20 } }))
+  // @ts-expect-error
+  expectType<{ a: string }>(useControls({ a: { value: 10, min: 0, max: 20 } }))
+})()
+
+/**
+ * strings
+ */
+;(() => {
+  expectType<{ a: string }>(useControls({ a: 'some string' }))
+  // @ts-expect-error
+  expectType<{ a: number }>(useControls({ a: 'some string' }))
+})()
+
+/**
+ * selects
+ */
+;(() => {
```

Comment:
There should be another check:

```ts
expectType<{ a: string | string[] | number }>(useControls({ a: { value: 3, options: ['foo', ['foo', 'bar']] } }))
```

In fact, the `select` object also accepts that a `value` property which type would extend `options`. Hope it makes sense?

Reply:
Awesome! Fixed in 9ec0645 🥳